torch/autograd/profiler.py

```diff
@@ -19,27 +19,72 @@ def __str__(self):
     def table(self, sort_by=None):
         return build_table(self, sort_by)
 
+    def export_chrome_trace(self, path):
+        """Exports an EventList as a Chrome tracing tools file.
+
+        The checkpoint can be later loaded and inspected under ``chrome://tracing`` URL.
+
+        Arguments:
+            path (str): Path where the trace will be written.
+        """
+        import json
+        with open(path, 'w') as f:
+            chrome_events = []
+            for evt in self:
+                chrome_events.append(dict(
+                    name=evt.name,
+                    ph='X',
+                    ts=evt.start / 1000,
+                    dur=evt.cpu_time_total / 1000,
+                    tid='Autograd functions',
+                    pid='Autograd functions',
+                    args={},
+                ))
+            json.dump(chrome_events, f)
+
+    def key_averages(self):
+        """Averages all function events over their keys.
+
+        Returns:
+            An EventList containing FunctionEventAvg objects.
+        """
+        stats = defaultdict(FunctionEventAvg)
+        for evt in self:
+            stats[evt.key] += evt
+        return EventList(stats.values())
+
+    def total_average(self):
+        """Averages all events.
+
+        Returns:
+            A FunctionEventAvg object.
+        """
+        total_stat = FunctionEventAvg()
+        for evt in self:
+            total_stat += evt
+            total_stat.key = None
+        total_stat.key = 'Total'
+        return total_stat
+
 
 class profile(object):
     """Context manager that manages autograd profiler state and holds a summary of results.
 
     Arguments:
-        use_nvprof (bool, optional): If True, uses nvprof (might incur high overhead and
-            assumes that the whole process is running inside nvprof), otherwise uses a custom
-            CPU-only profiler (with negligible overhead). Default: False.
-        trace_path (str, optional): A path of the CUDA checkpoint. If specified, it will be left
-            unmodified after profiling finishes, so it can be opened and inspected in nvvp. Otherwise
-            it will be created in a temporary directory and removed after reading the results.
         enabled (bool, optional): Setting this to False makes this context manager a no-op.
             Default: True.
 
+    .. warning:
+        This context managers should not be called recursively, i.e. at most one
+        instance should be enabled at any given time.
+
     Example:
         >>> x = Variable(torch.randn(1, 1), requires_grad=True)
         >>> with torch.autograd.profiler.profile() as prof:
         ...     y = x ** 2
         ...     y.backward()
         >>> # NOTE: some columns were removed for brevity
-        ... print(prof.key_averages())
+        ... print(prof)
```

Comment:
Any particular reason this change was reverted?

Reply:
Yeah. The table doesn't show key averages (PowConstant appears twice)

Reply:
Can we delete the duplicate PowConstant entry and then suggest `key_averages`? I have found that `key_averages` is ~essentially what you want for the first try, not the big dump.